### PR TITLE
Improve i18n support

### PR DESCRIPTION
**Pull request by jcfr**
This PR supersedes https://github.com/Slicer/SlicerGitSVNArchive/pull/1162 and should be consolidated with work contributed in https://github.com/Slicer/Slicer/pull/5021

### CODE REVIEW

**Comment by lassoan** [comment]
@jcfr this is great. Would you be able to join the weekly Slicer i18n meeting tomorrow at 9am to discuss with @pieper and the two computer science students who will work on this the design that we should be aiming for? If tomorrow does not work then another Wednesday at 9am will work, too.

File: Modules/Scripted/Endoscopy/Endoscopy.py [hunk 1]
```diff
@@ -63,7 +65,7 @@ def setup(self):
 
     # Path collapsible button
     pathCollapsibleButton = ctk.ctkCollapsibleButton()
-    pathCollapsibleButton.text = "Path"
+    pathCollapsibleButton.text = tr("EndoscopyWidget", "Path")
```
Comment (pieper) [inline Modules/Scripted/Endoscopy/Endoscopy.py:68]: Would be really nice if we could have a decorator to supply the translation context just to make things less verbose.

**Comment by lassoan** [comment]
Just got the info that tomorrow does not work for the students. @jcfr let us know if next Wednesday (Dec 7) 9am works for you.

**Comment by jcfr** [comment]
>  let us know if next Wednesday (Dec 7) 9am works for you.

That works for me.

**Comment by jcfr** [comment]
Closing.

The i18n support has been superseded by:
* Slicer specific pull-request associated with the [domain:i18n](https://github.com/Slicer/Slicer/pulls?q=is%3Apr++label%3Adomain%3Ai18n+-label%3Astatus%3Ainvalid+-label%3Astatus%3Awontfix+)
* Work done at https://github.com/Slicer/SlicerLanguagePacks

**Pull request closed without merging**